snapshots/devmapper/snapshotter.go

```diff
@@ -438,7 +457,7 @@ func (s *Snapshotter) buildMounts(snap storage.Snapshot) []mount.Mount {
 	mounts := []mount.Mount{
 		{
 			Source:  s.getDevicePath(snap),
-			Type:    fsTypeExt4,
+			Type:    s.config.FileSystemType,
```

Comment:
For the existing `ext4` snapshotter, the `buildMounts` returns wrong type if user try to use `xfs`. Not sure that we should persist the fstype in boltdb... or not allow user to use different fstypes

Reply:
This is an interesting question: We have couple of options
a. Allow snapshots with different fsTypes and returns only the ones that matches current fsType from running config
b. Don't allow snapshots with different fsType. For this we can check if config's fsType matches with the snaphot's fsType and return error if there is mismatch.

To know fsType of the snapshot, we can either save it in db or find out by other means from the snapshot.

Reply:
Implemented using labels.